src/espIdf/tracing/heapTraceManager.ts

```diff
@@ -0,0 +1,285 @@
+        super();
+        this.treeDataProvider = treeDataProvider;
+        this.archiveDataProvider = archiveDataProvider;
+        const tclConnectionParams = { host: "localhost", port: 6666 };
+        this.heapTraceNotificationTCLClientHandler = new TCLClient(tclConnectionParams);
+        this.heapTraceCommandChainTCLClientHandler = new TCLClient(tclConnectionParams);
+        this.heapTraceChannel = vscode.window.createOutputChannel("Heap Trace");
+    }
+
+    public async start() {
+        try {
+            if (await OpenOCDManager.init().promptUserToLaunchOpenOCDServer()) {
+                const commandChain = CommandChain.init();
+                commandChain.clear();
+                this.heapTraceChannel.clear();
+                this.showStopButton();
+                const workspace = vscode.workspace.workspaceFolders ?
+                    vscode.workspace.workspaceFolders[0].uri.fsPath : "";
+                if (!fileExists(join(workspace, "trace"))) {
+                    mkdirSync(join(workspace, "trace"));
+                }
+                const addresses = await this.getAddressFor(["heap_trace_start", "heap_trace_stop"]);
+                const fileName = `file://${join(workspace, "trace")}/htrace_${new Date().getTime()}.svdat`;
+                const target = idfConf.readParameter("idf.adapterTargetName");
+                let showStopButtonTimer: any;
+
+                commandChain
+                    .buildCommand({
+                        command: "reset halt",
+                        responseHandler: { type: "notificationResponse", handler: ["type target_reset mode halt", "type target_state state halted"] },
+                    })
+                    .buildCommand({
+                        command: `bp 0x${addresses.heap_trace_start} 4 hw`,
+                        responseHandler: { type: "commandResponse", handler: [] },
+                    })
+                    .buildCommand({
+                        command: `bp 0x${addresses.heap_trace_stop} 4 hw`,
+                        responseHandler: { type: "commandResponse", handler: [] },
+                    })
+                    .buildCommand({
+                        command: "resume",
+                        responseHandler: {
+                            type: "notificationResponse",
+                            handler: ["type target_event event resumed", "type target_state state running", "type target_state state halted"],
+                        },
+                    })
+                    .buildCommand({
+                        command: "reg pc", responseHandler: {
+                            type: "commandResponse",
+                            handler: [addresses.heap_trace_start],
+                        },
+                    })
+                    .buildCommand({
+                        command: `rbp 0x${addresses.heap_trace_start}`,
```

Comment:
Question, why do we need to set breakpoints on the current PC before/after the sysview command? (cc @gerekon)

Reply:
@igrr  This command removes breakpoint.